File: capacity-club-api/src/domain-modules/indoor-training/model/filter/indoor-training.filter.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsOptional } from 'class-validator';
import { Workout } from 'domain-modules/workout/model';

/**
 * Class used to filter indoor training.
 * This class defines the criteria used to filter a list of indoor training.
 */
export class IndoorTrainingFilter {
  @ApiProperty()
  @IsOptional()
  title: string;
  @ApiProperty()
  @IsOptional()
  training_date: Date;
  @ApiProperty()
  @IsOptional()
  start_hours: string;
  @ApiProperty()
  @IsOptional()
  end_hours: string;
  @ApiProperty()
  @IsOptional()
  nb_place: number;
  @ApiProperty()
  @IsOptional()
  nb_subscription: number;
  @ApiProperty()
  @IsOptional()
  workout: Workout;
}

File: capacity_club_mobile_app/lib/domain-features/common-features/data/model/filter/indoor_training_filter.dart
class IndoorTrainingFilter {
  final String? title;
  final DateTime? training_date;
  final String? start_hours;
  final String? end_hours;
  final int? nb_place;
  final int? nb_subscription;

  const IndoorTrainingFilter({
    this.title,
    this.training_date,
    this.start_hours,
    this.end_hours,
    this.nb_place,
    this.nb_subscription,
  });

  factory IndoorTrainingFilter.fromJson(Map<String, dynamic> json) {
    return IndoorTrainingFilter(
      title: json['title'],
      training_date: json[
          'training_date'], // training_date: json['training_date'] != null? DateTime.parse(json['training_date']): null,
      start_hours: json['start_hours'],
      end_hours: json['end_hours'],
      nb_place: json['nb_place'],
      nb_subscription: json['nb_subscription'],
    );
  }

  Map<String, dynamic> toJson() => {
        'title': title,
        'training_date':
            training_date, //'training_date': training_date?.toIso8601String(),
        'start_hours': start_hours,
        'end_hours': end_hours,
        'nb_place': nb_place,
        'nb_subscription': nb_subscription,
      };
}
